### PR TITLE
Logout button

### PR DESCRIPTION
**Pull request by kozaktar**
-Added a dropdown menu to the navbar
-added log out button to dropdown menu in the nav bar
-enabled user to always select google account and enter password on login
-enabled a setting in .env file to force user consent screen (in case you need to update app consents)

https://user-images.githubusercontent.com/20387499/106467067-23828780-646a-11eb-9462-aa9366e24e3d.mp4

### CODE REVIEW

File: client/src/components/UserMenu.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    fontWeight: "bold"
+  },
+  Menu: {
+    marginLeft: 25
+  },
+  icon: { marginRight: 10 }
+}));
+
+export default function UserMenu({ name }) {
+  // const [anchorEl, setAnchorEl] = React.useState(null);
```
Comment (dproc96) [inline client/src/components/UserMenu.js:41]: You can delete any commented out code before merging

File: server/utils/googleAuthUtils.js [hunk 2]
```diff
@@ -21,20 +25,26 @@ function authorize(token) {
...
 }
 
+function generateAuthUrl() {
+  const settings = {
+    access_type: "offline",
+    scope: SCOPES,
+    prompt: "select_account"
+  };
+
+  if(process.env.FORCE_CONSENT== 'true'){
```
Comment (dproc96) [inline server/utils/googleAuthUtils.js:35]: You should run prettier on this file before merging

**Review by dproc96 — APPROVED** [review]
Had a couple of minor comments to address before merging, but pending those this is good to go!

File: client/src/components/Header/NavBar.js [hunk 3]
```diff
@@ -93,14 +93,7 @@ export default function NavBar() {
...
         </Box>
-
-        <Link
-          variant="subtitle1"
-          onClick={preventDefault}
-          className={classes.link}
-        >
-          John Doe
-        </Link>
+        <UserMenu name="John Doe"/>
```
Comment (mattchx) [inline client/src/components/Header/NavBar.js:96]: You can integrate context here when you merge dev into your branch.

File: client/src/components/UserMenu.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  // };
+
+  const classes = useStyles();
+  const history = useHistory();
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef(null);
+  const setAuthenticated = useSetAuthenticated();
+
+  const handleToggle = () => {
+    setOpen(prevOpen => !prevOpen);
```
Comment (mattchx) [inline client/src/components/UserMenu.js:58]: Would there be any difference between `setOpen(prevOpen => !prevOpen)` and `setOpen(!prevOpen)`

File: client/src/components/UserMenu.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  };
+
+  function handleListKeyDown(event) {
+    if (event.key === "Tab") {
+      event.preventDefault();
+      setOpen(false);
+    }
+  }
+
+  // return focus to the button when we transitioned from !open -> open
```
Comment (mattchx) [inline client/src/components/UserMenu.js:75]: Which button would that be in this case?

File: client/src/components/UserMenu.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        {({ TransitionProps, placement }) => (
+          <Grow
+            {...TransitionProps}
+            style={{
+              transformOrigin:
+                placement === "bottom" ? "center top" : "center bottom"
+            }}
+          >
+            <Paper className={classes.Menu}>
+              <ClickAwayListener onClickAway={handleClose}>
```
Comment (mattchx) [inline client/src/components/UserMenu.js:120]: Nice implementation!

**Review by mattchx — APPROVED** [review]
Well done! The styling looks great.

File: client/src/components/UserMenu.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        disablePortal
+      >
+        {({ TransitionProps, placement }) => (
+          <Grow
+            {...TransitionProps}
+            style={{
+              transformOrigin:
+                placement === "bottom" ? "center top" : "center bottom"
+            }}
+          >
```
Comment (sina-jamshidi) [inline client/src/components/UserMenu.js:118]: Nice detail with the transition

**Review by sina-jamshidi — APPROVED** [review]

**Pull request merged**